### PR TITLE
VBLOCKS-2504: Updated to use system call notifications

### PR DESCRIPTION
**Pull request by afalls-twilio**
## Submission Checklist

 - [x] Updated the `CHANGELOG.md` to reflect any **feature**, **bug fixes**, or **known issues** made in the source code
 - [x] Tested code changes and observed expected behavior in the example app
 - [x] Performed a visual inspection of the `Files changed` tab prior to submitting the pull request for review to ensure proper usage of the style guide

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

## Description

Updated to use system provided incoming and ongoing call notifications, introduced in API 34, made backwards compatible by AndroidX.

## Breakdown

- Removed custom notifications & replaced with system style based notifications
- Removed prefix 'client:' from the incoming/ongoing call name

## Validation

- Tested incoming & ongoing calls on both android 34 and android 29 level devices

## Additional Notes

None

### CODE REVIEW

**Comment by afalls-twilio** [comment]
Incoming call
Screenshot 2024-02-13 at 3 38 29 PM

Ongoing call
Screenshot 2024-02-13 at 3 38 40 PM

Ongoing call (Android 29)
Screenshot 2024-02-13 at 3 40 09 PM

File: CHANGELOG.md [hunk 1]
```diff
@@ -23,6 +23,8 @@ Twilio Voice React Native SDK has now reached milestone `beta.5`. Included in th
 
 #### Android
 - Call timestamp now in simplified ISO-8601 format, not stored as a double from epoch.
+- Now using system provided notification styles for incoming & ongoing calls. This insures visual consistnacy between devices.
```
Comment (ramapal) [inline CHANGELOG.md:26]: typo - `consistnacy`

Comment (ramapal) [reply CHANGELOG.md:26]: Minor - Do we need `Now`? Suggestion: `- Uses system provided notification styles ...`

**Review by ramapal — APPROVED** [review]
A couple of comments about the changelog. Took a cursory pass and changes LGTM. Learnt a couple of things about Android Notifications from this PR.

File: android/src/main/java/com/twiliovoicereactnative/NotificationUtility.java [hunk 2]
```diff
@@ -70,39 +72,32 @@ public static Notification createIncomingCallNotification(@NonNull Context conte
...
-        .setCategory(Notification.CATEGORY_CALL)
-        .setAutoCancel(true)
-        .setCustomContentView(remoteViews)
-        .setCustomBigContentView(remoteViews)
-        .setContentIntent(piForegroundIntent);
-    if (fullScreenIntent) {
-      builder.setFullScreenIntent(piForegroundIntent, true);
-    }
-    return builder.build();
+    return  constructNotificationBuilder(context, channelImportance)
```
Comment (mhuynh5757) [inline android/src/main/java/com/twiliovoicereactnative/NotificationUtility.java:75]:
```suggestion
    return constructNotificationBuilder(context, channelImportance)
```

File: android/src/main/java/com/twiliovoicereactnative/NotificationUtility.java [hunk 3]
```diff
@@ -70,39 +72,32 @@ public static Notification createIncomingCallNotification(@NonNull Context conte
...
+      .setStyle(NotificationCompat.CallStyle.forIncomingCall(
+        incomingCaller, piRejectIntent, piAcceptIntent))
+      .build();
   }
 
   public static Notification createCallAnsweredNotificationWithLowImportance(@NonNull Context context,
                                                                              @NonNull final CallRecord callRecord) {
     final int smallIconResId = getSmallIconResource(context);
-    final String title = getDisplayName(callRecord.getCallInvite());
+    final Person activeCaller =  new Person.Builder()
```
Comment (mhuynh5757) [inline android/src/main/java/com/twiliovoicereactnative/NotificationUtility.java:91]:
```suggestion
    final Person activeCaller = new Person.Builder()
```

File: android/src/main/java/com/twiliovoicereactnative/NotificationUtility.java [hunk 4]
```diff
@@ -111,26 +106,24 @@ public static Notification createCallAnsweredNotificationWithLowImportance(@NonN
...
+      .setOngoing(true)
+      .addPerson(activeCaller)
+      .setStyle(NotificationCompat.CallStyle.forOngoingCall(activeCaller, piEndCallIntent))
       .build();
   }
 
   public static Notification createOutgoingCallNotificationWithLowImportance(@NonNull Context context,
                                                                              @NonNull final CallRecord callRecord) {
     final int smallIconResId = getSmallIconResource(context);
+    final Person activeCaller =  new Person.Builder()
```
Comment (mhuynh5757) [inline android/src/main/java/com/twiliovoicereactnative/NotificationUtility.java:124]:
```suggestion
    final Person activeCaller = new Person.Builder()
```

File: android/src/main/java/com/twiliovoicereactnative/NotificationUtility.java [hunk 5]
```diff
@@ -111,26 +106,24 @@ public static Notification createCallAnsweredNotificationWithLowImportance(@NonN
...
+      .setName("outgoing call")
```
Comment (mhuynh5757) [inline android/src/main/java/com/twiliovoicereactnative/NotificationUtility.java:125]: Can we expose this to the JS layer when the JS layer tries to make a call? Is there any utility to that?

**Review by mhuynh5757 — CHANGES_REQUESTED** [review]

File: android/src/main/java/com/twiliovoicereactnative/NotificationUtility.java (hunk 5 above)
Comment (afalls-twilio) [reply android/src/main/java/com/twiliovoicereactnative/NotificationUtility.java:125]: not with this pr..  when we discus allowing for users to provide custom  notifications, we can revisit it.

**Review by mhuynh5757 — APPROVED** [review]

**Pull request merged**